modules/data-entry/src/main/frontend/src/questionnaire/DateQuestionFull.jsx

```diff
@@ -0,0 +1,194 @@
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useState } from "react";
+
+import { TextField, Typography, withStyles } from "@material-ui/core";
+
+import moment from "moment";
+import * as jdfp from "moment-jdateformatparser";
+
+import Answer from "./Answer";
+import Question from "./Question";
+import QuestionnaireStyle from "./QuestionnaireStyle";
+
+import AnswerComponentManager from "./AnswerComponentManager";
+import DateQuestionUtilities from "./DateQuestionUtilities";
+
+// Component that renders a date/time question
+// Selected answers are placed in a series of <input type="hidden"> tags for
+// submission.
+//
+// Optional props:
+// text: the question to be displayed
+// type: "timestamp" for a single date or "interval" for two dates
+// dateFormat: A string specifying a date format including date, as detected by DateQuestionUtilities
+// lowerLimit: lower date limit (inclusive) given as an object or string parsable by moment()
+// upperLimit: upper date limit (inclusive) given as an object or string parsable by moment()
+// Other options are passed to the <question> widget
+//
+// Sample usage:
+//<DateQuestion
+//  text="Please enter a date-time in 2019"
+//  dateFormat="yyyy-MM-dd hh:mm:ss"
+//  lowerLimit={new Date("01-01-2019")}
+//  upperLimit={new Date("12-31-2019")}
+//  type="timestamp"
+//  />
+function DateQuestionFull(props) {
+  let {existingAnswer, classes, ...rest} = props;
+  let {text, dateFormat, minAnswers, type, lowerLimit, upperLimit} = {dateFormat: "yyyy-MM-dd", minAnswers: 0, type: DateQuestionUtilities.TIMESTAMP_TYPE, ...props.questionDefinition, ...props};
+
+  let startValues = existingAnswer && existingAnswer[1].value || "";
+
+  let stripTimeZone = (dateString) => {
+    // Remove the time zone (eg. "-5:00") from the end of a sling provided date string
+    return dateString.replace(/-[0-9]{2}:[0-9]{2}$/gm, '');
+  }
+
+  const [ startDate, setStartDate ] = useState(DateQuestionUtilities.amendMoment(
+    stripTimeZone(typeof(startValues) === "object" ? startValues[0] : startValues),
+    DateQuestionUtilities.slingDateFormat)
+  );
+  const [ endDate, setEndDate ] = useState(
+    typeof(startValues) === "object"
+    ? DateQuestionUtilities.amendMoment(stripTimeZone(startValues[1]), DateQuestionUtilities.slingDateFormat)
+    : null
+  );
+  const upperLimitMoment = upperLimit ? DateQuestionUtilities.amendMoment(upperLimit, DateQuestionUtilities.slingDateFormat) : null;
```

Comment:
This is repeated several times. Let's make it shorter:
* `DateQuestionUtilities.amendMoment` should return `null` if the first parameter is `null`
* If the second (format) parameter is missing, ` DateQuestionUtilities.slingDateFormat` should be used by default
```suggestion
  const upperLimitMoment = DateQuestionUtilities.amendMoment(upperLimit);
```